target/linux/realtek/image/rtl838x.mk

```diff
@@ -23,7 +23,8 @@ define Device/d-link_dgs-1210
   CAMEO_BOARD_VERSION := 32
   IMAGES += factory_image1.bin
   IMAGE/factory_image1.bin := append-kernel | pad-to 64k | \
-        append-rootfs | pad-rootfs | pad-to 16 | check-size | dlink-headers
+	append-rootfs | pad-rootfs | pad-to 16 | check-size | \
+	dlink-version | dlink-headers
```

Comment:
Looking at a FW image for the [DGS-1210-28](https://support.dlink.com/resource/PRODUCTS/DGS-1210-SERIES/REVF/FIRMWARE/v6.30.016/DGS-1210-28_REVF_FIRMWARE_v6.30.016.zip), the firmware version string + CAMEOTAG are 32 bytes. So I assume that's why you're adding "OpenWrt" + 25 bytes here.

Since the version string is written to flash, it should also be covered by `check-size`. `dlink-version` will only add 32 bytes, at a point where you're only 16 bytes into the next erase block, so the result should never be different though.

The `dlink-version` tag is appended to the data after the DEADCODE marker. So it doesn't look like this will survive the jffs2 initialisation for OpenWrt's rootfs_data partition. Is this something you wish to support?